test/jdk/java/net/httpclient/HttpRequestNewBuilderTest.java

```diff
@@ -294,53 +306,70 @@ public void testUserDefinedMethod(HttpRequest request) {
     }
 
     @Test(dataProvider = "testRequests")
-    public void testHeaders(HttpRequest request) {
+    public void testAddHeader(HttpRequest request) {
         BiPredicate<String, String> filter = (n, v) -> true;
-        var r = HttpRequest.newBuilder(request, filter).headers("newName", "newValue").build();
 
+        var r = HttpRequest.newBuilder(request, filter).headers("newName", "newValue").build();
         assertEquals(r.headers().firstValue("newName").get(), "newValue");
         assertEquals(r.headers().allValues("newName").size(), 1);
         assertAllOtherElementsEqual(r, request, "headers");
     }
 
     @Test(dataProvider = "testRequests")
     public void testRemoveHeader(HttpRequest request) {
-        BiPredicate<String, String> filter = (n, v) -> !n.equalsIgnoreCase("testName");
+        BiPredicate<String, String> filter = (n, v) -> !n.equalsIgnoreCase("testName1");
```

Comment:
This assumes that there is at least one `request` such that `request.headers().firstValue("testName1").isPresent()` otherwise the test will trivially pass.
I wonder if we need a specific dataProvider here, so that we can assert:

```
assertTrue(request.headers().firstValue("testName1").isPresent());
```

first thing in the test? That - or have an AtomicInteger that counts how many time the original request has the conf we want to remove, and then assert that the counter > 0 in an `@AfterClass` method?

Reply:
Would it be enough to check in each test case that ensures if we are examining a request with headers, it contains at least one called `testName1`? i.e.
```
        if(!request.headers().map().isEmpty())
            assertTrue(request.headers().map().containsKey("testName1"));
```

Reply:
I will settle for that 👍

Reply:
OK. Check added to commit https://github.com/openjdk/jdk/pull/1059/commits/68ea89e1c55735adc9230b8abc07e1df3ffc03ca